mustang_codegen/lib/src/batch_generator/app_batch_generator.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:async';
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:build/build.dart';
+import 'package:glob/glob.dart';
+import 'package:mustang_codegen/src/batch_generator/batch_model.dart';
+import 'package:mustang_codegen/src/batch_generator/enum_param_visitor.dart';
+import 'package:mustang_core/mustang_core.dart';
+import 'package:source_gen/source_gen.dart';
+
+class AppBatchGenerator implements Generator {
+  @override
+  FutureOr<String?> generate(LibraryReader library, BuildStep buildStep) async {
+    Iterable<AnnotatedElement> annotatedElements =
+        library.annotatedWith(TypeChecker.fromRuntime(Batch));
+
+    StringBuffer appBatchBuffer = StringBuffer();
+
+    String res = await _generateBatch(buildStep, annotatedElements.first);
+    appBatchBuffer.writeln(res);
+
+    return '$appBatchBuffer';
+  }
+
+  Future<String> _generateBatch(
+    BuildStep buildStep,
+    AnnotatedElement annotedElement,
+  ) async {
+    EnumElement batch = annotedElement.element as EnumElement;
+    List<BatchModel> methodsForBatch = [];
```

Comment:
`batchModels` may be appropriate name here. I was expecting to see something about the method here but turned out to be a model.